osu.Game/Graphics/Containers/StoryboardContainer.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Game.Storyboards;
+using osu.Game.Storyboards.Drawables;
+
+namespace osu.Game.Graphics.Containers
+{
+    /// <summary>
+    /// A container that handles <see cref="Storyboard"/> loading, as well as applies user-specified visual settings to it.
+    /// </summary>
+    public class StoryboardContainer : UserDimContainer
+    {
+        private readonly Storyboard storyboard;
+        private DrawableStoryboard drawableStoryboard;
+
+        public StoryboardContainer(Storyboard storyboard)
+        {
+            this.storyboard = storyboard;
+            EnableUserDim.Default = true;
+            EnableUserDim.Value = true;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            initializeStoryboard(false);
+        }
+
+        protected override void LoadComplete()
+        {
+            ShowStoryboard.BindValueChanged(_ => initializeStoryboard(true), true);
+            base.LoadComplete();
+        }
+
+        protected override void ApplyFade()
```

Comment:
having overrides without calling `base` is a usually a bad sign.

at this point, I'd prefer you made UserDimContainer abstract and has `DimmableStoryboardContainer` and `DimmableBackgroundContainer` (and move `Background` there so it doesn't even exist in the DimmableStoryboardContainer).